receiver/postgresqlreceiver/scraper_test.go

```diff
@@ -263,6 +263,13 @@ func (m *mockClient) initMocks(database string, databases []string, index int) {
 				replayLag:    700,
 				writeLag:     800,
 			},
+			{
```

Comment:
Would this test here yield the same errors from the reported issue when the fix isn't in place?

Reply:
I don't think so. The error came from within the client library. This is using a mock client so is only testing part of the change. Unfortunately, I think a proper test would require configuring replication within an integration test, which is a bit outside of my wheelhouse.

Reply:
That's fine, I think it's close enough.